### PR TITLE
Adding long lived SA token option (#917)

### PR DESCRIPTION
**Pull request by gillcaleb**
Addresses #917

The intent of this PR is to allow for the [creation of a long-lived API token for the server SA](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#create-token).

### CODE REVIEW

**Comment by gillcaleb** [comment]
It's unclear to my why the license/cla check continues to remain in a pending state as I've gone through the signing process and verified its completion on my end. I've tried re-checking it numerous times but to no avail.

**Comment by stavvy-cgill** [comment]
> > It's unclear to my why the license/cla check continues to remain in a pending state as I've gone through the signing process and verified its completion on my end. I've tried re-checking it numerous times but to no avail.
> 
> Usually that happens if your contributions are under a different account, ie, email. For example, I see that @stavvy-cgill is making the comments, but @gillcaleb is the commenter here. Try signing with your @stavvy-cgill account and that should fix it. Thanks!

Thank you!

**Comment by gillcaleb** [comment]
Hi team! Just wanted to check back in to get a sense for when this might get a review. Thanks!

File: Chart.yaml
Comment (tomhjp) [inline Chart.yaml:6]: Please can you revert this? We make the call on what the next version will be at release time, so no need to edit it in the meantime. Thanks!

File: values.yaml
Comment (tomhjp) [inline values.yaml:922]: I'd like us to be very explicit that this is not recommended:

```suggestion
    # Create a Secret API object to store a non-expiring token for the service account.
    # Prior to v1.24.0, Kubernetes used to generate this secret for each service account by default.
    # Kubernetes now recommends using short-lived tokens from the TokenRequest API or projected volumes instead if possible.
    # For more details, see https://kubernetes.io/docs/concepts/configuration/secret/#service-account-token-secrets
    # serviceAccount.create must be equal to 'true' in order to use this feature.
    createSecret: false
```

I've also suggested a slightly different name, but I won't pepper the accompanying suggestions through the rest of the PR to avoid spam.

File: templates/server-serviceaccount-token.yaml
Comment (tomhjp) [inline templates/server-serviceaccount-token.yaml:1]: Please can we rename the file to `server-serviceaccount-secret.yaml`? The `metadata.name` field makes sense to stick with -token, but I think we're fairly consistent about using the API object name in the file name.

**Review by tomhjp — COMMENTED** [review]
Thanks for raising this - I'm ok with adding it but I want to make sure Kubernetes' alternative recommendations are made clear. LGTM once a few naming and documentation comments are addressed. Thanks for including the tests and schema update too!

**Comment by gillcaleb** [comment]
> Thanks for raising this - I'm ok with adding it but I want to make sure Kubernetes' alternative recommendations are made clear. LGTM once a few naming and documentation comments are addressed. Thanks for including the tests and schema update too!

Appreciate the feedback! I went ahead and added the suggested changes.

**Comment by kaibepperling** [comment]
Thanks for opening the PR @gillcaleb! We ran in to the same thing today and were about to open a PR. 
Is anything blocking a release of that change?

File: templates/_helpers.tpl
Comment (tomhjp) [inline templates/_helpers.tpl:92]: To match the values.yaml file, here and elsewhere:

```suggestion
    (eq (.Values.server.serviceAccount.createSecret | toString) "true")) -}}
```

**Review by tomhjp — COMMENTED** [review]
Sorry for the delay revisiting this - and thanks for addressing the comments. The changes look good to me, except we just need a very mechanical replacement of `generateSecret` with `createSecret` in a few places to match up with the latest name in values.yaml.

**Review by tomhjp — APPROVED** [review]
LGTM, thanks! I'll wait for the acceptance tests to pass and then merge.

**Pull request merged**
